### PR TITLE
Error reporting to server

### PR DESCRIPTION
**Pull request by camalouu**
Added functionality to send errors.log file to server when an exception occurs.

### CODE REVIEW

**Comment by JeongJun-Lee** [comment]
When you make a pr, you have to have your commit only without other's commits. For that you need to create your own work branch for your work only and create a pr from the branch. And because this commit is not for general use, pls make the pr to the globalrank-restful branch, not the dev branch.

File: OpenTyping/App.config [hunk 1]
```diff
@@ -41,5 +41,10 @@
       </dependentAssembly>
     </assemblyBinding>
   </runtime>
+	<system.net>
```
Comment (JeongJun-Lee) [inline OpenTyping/App.config:44]: why should this setting be added to the App config?

File: OpenTyping/ErrorReport.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace OpenTyping
+{
+    class ErrorReport
+    {
+        public static readonly string errorFile = "./errors.log";
+        public static readonly string errorServerUri = "http://DESKTOP-OPL53JG:30001/errors";
+        public static void SendErrorLogs()
+        {
+            if (File.Exists(errorFile))
```
Comment (JeongJun-Lee) [inline OpenTyping/ErrorReport.cs:18]: The role of a file handling like checking if the file is exist is not and deleting a file this class.  I'll refactor the RestfulProvider class(https://github.com/roboticsware/OpenTyping/blob/globalrank-restful/OpenTyping/RestfulProvider.cs) from WebClient to HttpClient and add new API to report error to the class. After finishing the work, I'll let you know. Then you can just utilize the API for sending the prepared reformatted error with additional info to the server.

**Review by JeongJun-Lee — CHANGES_REQUESTED** [review]

File: OpenTyping/ErrorReport.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                {
+                    using (WebClient wc = new WebClient())
+                    {
+                        var os = System.Environment.OSVersion;
+                        var version = Assembly.GetEntryAssembly().GetName().Version;
+
+                        string platformInfo =
+                            $"\nApp version: {version}\n" +
+                            $"Platform: {os.Platform}\n" +
+                            $"OS version string: {os.VersionString}\n";
```
Comment (JeongJun-Lee) [inline OpenTyping/ErrorReport.cs:29]: is os version in the platform info different from the value of os variable?

File: OpenTyping/App.config (hunk 1 above)
Comment (camalouu) [reply OpenTyping/App.config:44]: It removes "Expect: 100-continue" header. I don't know why but for some reason program does not send the error file even if it receives the response with 100 status code.

**Pull request closed without merging**